### PR TITLE
Fixes #15

### LINKED ISSUE #15: Support for defining data sources to be referenced by queries
hi
may be you could enable default config for
db connection (per file may be), driver and interval

as sometime we have lots of queries to one db
like a long list
and have to add all those lines each time

thanks

### PR DESCRIPTION
**Pull request by haxorof**
I have done an implementation but I must say that I am new to golang and come from a C++/Java background so please tell me to fix stuff if I done something which can be done in a better way etc. 😄 

I have executed some manual test to verify that the existing stuff also works to not break compatibility.

### CODE REVIEW

**Comment by bruth** [comment]
Thanks @haxorof for putting this together. After my first pass I am not sure I like having four additional libraries introduced for a fairly simple set of validation logic (I tend to prefer limiting dependencies in general). Having a single function that takes the config and checks here it needs to and falling back to defaults is often sufficient and more clear since it is just Go rather than declarative rules in the struct tags.

**Comment by haxorof** [comment]
Thanks for the feedback @bruth! I will rework it a bit to reduce dependencies etc. 
I will push some new commit and let you know when I done that. 😄

**Comment by haxorof** [comment]
Now I removed the validation dependencies and hope it is closer to what you except it to be.

File: config.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+}
+
+func loadQueriesInDir(path string, config *Config) (QueryList, error) {
+	log.Printf("Load queries from directory [%s]", path)
+	queries := make(QueryList, 0)
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+	for _, f := range files {
```
Comment (bruth) [inline config.go]: Add an empty line above to split up the first part of the function. As a general aesthetic, I tend to break up chunks of group based on obvious relatedness or when an error needs to be handled. So in this case the above error is handled, so an empty line can go below. Likewise after `for`, `select`, and `switch` blocks I tend to add a line below even if the next statement is a `return`.

File: config_test.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
+package main
+
+import (
+	"reflect"
+	"testing"
+	"time"
+	//"github.com/davecgh/go-spew/spew"
```
Comment (bruth) [inline config_test.go]: Remove since it is commented out?

File: config.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+		QueryTimeout:      DefaultTimeout,
+		QueryValueOnError: "",
+	}
+}
+
+func newConfig() *Config {
+	return &Config{Defaults: createDefaultsData()}
+}
+
+func appendDefaults(c Config) Config {
```
Comment (haxorof) [inline config.go]: Do not remember why exactly now. 😄 Will change to pointer.

File: config.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+	// Read queries for request body.
+	file, err := os.Open(queriesFile)
+	if err != nil {
+		return nil, fmt.Errorf("Error opening queries file: %s", err)
+	}
+	defer file.Close()
+	return decodeQueries(file, config)
+}
+
+func useFallbackIfEmpty(value string, fallback string) string {
```
Comment (haxorof) [inline config.go]: Will remove

File: config.go [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+	if value == "" {
+		return fallback
+	}
+	return value
+}
+
+func useFallbackIfZero(value time.Duration, fallback time.Duration) time.Duration {
```
Comment (haxorof) [inline config.go]: Will remove

File: config.go (hunk 1 above)
Comment (haxorof) [inline config.go]: Thanks! It is good feedback to hear, will think about this next time. Also thinking if some of these general aesthetic "rules" shall be written in some kind of code guideline markdown file just for it to be more clear for me and others in the future.

File: config_test.go (hunk 2 above)
Comment (haxorof) [inline config_test.go]: Will remove, forgot to remove it.

**Comment by haxorof** [comment]
@bruth: Ready for another round. 😄 Hopefully I understood your comments correctly.

**Comment by bruth** [comment]
Nice, you understood 😄

**Comment by bruth** [comment]
Feel free to squash and merge when you are ready.

**Pull request merged**